### PR TITLE
typescript enforced + clean devdepencies

### PR DESCRIPTION
**Pull request by hugoblanc**
I'll understand if you reject my PR but at least take a look. They are several ways to improve typescript usage.

### CODE REVIEW

File: src/utils/StorageHelper.ts [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 import * as ConfigStore from 'configstore'
+import { IDeployedDirectory, IMachine } from '../models/storage/StoredObjects'
 import Utils from './Utils'
 
 const CAP_MACHINES = 'CapMachines'
 const DEPLOYED_DIRS = 'DeployedDirs'
 
 export default class StorageHelper {
-    static instance: StorageHelper
+    public static instance: StorageHelper
```
Comment (githubsaturn) [inline src/utils/StorageHelper.ts]: Should be private

**Comment by hugoblanc** [comment]
I agree with you. A lot of these changes are not necessarily. But, that's one of the side effects of the linter. 
Maybe we could try to  elaborate a good tslint config (based on what you would like to ensure and what you wouldn't) ..

**Comment by githubsaturn** [comment]
> While others, such as changing function() to ()=>, explicitly adding public keyword (as opposed to public being default), and the extra comma at the end of each objects are just opinionated styling which makes this PR unnecessarily too large to be reviewed.

Except the ones that I pointed out above, the other enforcement changes are good. If you want, you can update this PR. Or we can abondon this PR and you can submit a new PR. Or I'll cherry-pick some changes from this PR manually.

**Comment by hugoblanc** [comment]
I should create a new one to have cleaner git history.A bit busy by now, I'll do it in one or two weeks.

**Comment by githubsaturn** [comment]
Sounds good. Take your time.

**Comment by hugoblanc** [comment]
[PR](https://github.com/caprover/caprover-cli/pull/45)

**Pull request closed without merging**